### PR TITLE
pkg: Use strings.Builder instead of bytes.Buffer where possible

### PR DESCRIPTION
**Pull request by twpayne**
strings.Builder is optimized for building strings and performs a little
better than bytes.Buffer.

### CODE REVIEW

**Review by christarazi — APPROVED** [review]
Nice! I didn't know `strings.Builder` existed :rocket:! Do you happen to have numbers or a percentage improvement while working on this? (No need to produce them, if not; more curious if anything.)

File: pkg/command/output.go [hunk 1]
```diff
@@ -86,13 +86,13 @@ func DumpJSONToSlice(data interface{}, jsonPath string) ([]byte, error) {
...
-	buf := new(bytes.Buffer)
-	if err := parser.Execute(buf, data); err != nil {
+	var sb strings.Builder
+	if err := parser.Execute(&sb, data); err != nil {
 		fmt.Fprintf(os.Stderr, "Couldn't parse jsonpath expression: '%s'\n", err)
 		return nil, err
 
 	}
-	return buf.Bytes(), nil
+	return []byte(sb.String()), nil
```
Comment (tklauser) [inline pkg/command/output.go:95]: It seems all users of this function convert the return value directly to a string before using it further. So we could even change the signature to:

```go
func DumpJSONToString(data interface{}, jsonPath string) (string, error)
```

and adjust the two callers accordingly. Might be better suited for a follow-up PR though.

**Comment by twpayne** [comment]
> Do you happen to have numbers or a percentage improvement while working on this? (No need to produce them, if not; more curious if anything.)

I think the benchmarks are very dependent on the use case. The main interest in `strings.Builder` seems to be the reduced number of memory allocations and copies.

**Review by kkourt — APPROVED** [review]

**Review by jrajahalme — APPROVED** [review]

**Review by joestringer — APPROVED** [review]

**Review by aanm — APPROVED** [review]

**Comment by twpayne** [comment]
test-me-please

**Comment by twpayne** [comment]
CI failures are because we've hit the Docker image pull limit 😢

**Comment by christarazi** [comment]
test-me-please

**Comment by aanm** [comment]
retest-gke

**Comment by aanm** [comment]
retest-net-next

**Comment by aanm** [comment]
retest-4.19

**Comment by aanm** [comment]
retest-4.9

**Comment by christarazi** [comment]
Looks like this PR needs a rebase.

**Comment by twpayne** [comment]
> Looks like this PR needs a rebase.

Ah, the constantly-moving target which is `master`...

**Comment by twpayne** [comment]
test-me-please

**Comment by twpayne** [comment]
retest-gke

**Comment by twpayne** [comment]
retest-4.9

**Comment by christarazi** [comment]
retest-4.9

**Comment by christarazi** [comment]
retest-gke

**Review by sayboras — APPROVED** [review]

File: pkg/command/output.go (hunk 1 above)
Comment (twpayne) [reply pkg/command/output.go:95]: Done in https://github.com/cilium/cilium/pull/14041.

**Pull request merged**